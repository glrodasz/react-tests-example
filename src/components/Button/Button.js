import React from "react";
import PropTypes from "prop-types";
import getClasses from "../../utils/getClasses";

const Button = ({ children, color, onClick }) => (
  <button
    className={getClasses(color)}
    onClick={onClick}
  >
    {children}
  </button>
);

Button.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired,
};

Button.defaultProps = {
  onClick: () => {},
  color: 'red',
}

export default Button;
