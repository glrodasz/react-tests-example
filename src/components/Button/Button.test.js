import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Button from "./Button";

describe("[ components / Button ]", () => {
  describe("#structure", () => {
    describe("when default props are provided", () => {
      it("should render correctly", () => {
        // given
        const props = {};

        // when
        const tree = renderer
          .create(<Button {...props}>button</Button>)
          .toJSON();

        // then
        expect(tree).toMatchSnapshot();
      });
    });

    describe("when `color` prop is `blue`", () => {
      it("should render correctly", () => {
        // given
        const props = { color: "blue" };

        // when
        const tree = renderer
          .create(<Button {...props}>button</Button>)
          .toJSON();

        // then
        expect(tree).toMatchSnapshot();
      });
    });

    describe("when `color` prop is `yellow`", () => {
      it("should render correctly", () => {
        // given
        const props = { color: "yellow" };

        // when
        const tree = renderer
          .create(<Button {...props}>button</Button>)
          .toJSON();

        // then
        expect(tree).toMatchSnapshot();
      });
    });
  });

  describe("#behaviour", () => {
    describe("when `Button` is clicked", () => {
      it("should call `onClick` handler", () => {
        // given
        const onClickMock = jest.fn();
        const wrapper = shallow(<Button onClick={onClickMock}>button</Button>);

        // when
        wrapper.simulate("click");

        // then
        expect(onClickMock).toHaveBeenCalledTimes(1);
      });
    });
  });
});
