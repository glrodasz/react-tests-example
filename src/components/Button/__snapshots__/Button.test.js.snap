// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ components / Button ] #structure when \`color\` prop is \`blue\` should render correctly 1`] = `
<button
  className="mt-2 relative bg-blue-500 text-white py-4 px-10 rounded text-2xl font-bold overflow-visible"
  onClick={[Function]}
>
  button
</button>
`;

exports[`[ components / Button ] #structure when \`color\` prop is \`yellow\` should render correctly 1`] = `
<button
  className="mt-2 relative bg-yellow-500 text-white py-4 px-10 rounded text-2xl font-bold overflow-visible"
  onClick={[Function]}
>
  button
</button>
`;

exports[`[ components / Button ] #structure when default props are provided should render correctly 1`] = `
<button
  className="mt-2 relative bg-red-500 text-white py-4 px-10 rounded text-2xl font-bold overflow-visible"
  onClick={[Function]}
>
  button
</button>
`;
